src/app/utils/dataLoaderUtil.js

```diff
@@ -1,156 +1,87 @@
+    path: 'hold/request',
+    params: '/:bibId-:itemId',
   },
-});
-
-const matchingPathData = (location) => {
-  const {
-    pathname,
-    search,
-  } = location;
+  home: {
+    action: resetState,
+    path: '',
+  },
+};
 
-  return pathInstructions
-    .map(instruction => ({
-      matchData: (pathname + search).match(instruction.expression),
-      pathType: instruction.pathType,
-    }))
-    .find(pair => pair.matchData)
-    || { matchData: null, pathType: null };
+// A simple function for loading data into the store. The only reason it is broken
+// out separately is because it is used front-end and back-end
+const successCb = (pathType, dispatch) => (response) => {
+  dispatch(routes[pathType].action(response.data));
 };
 
-function loadDataForRoutes(location, req, routeMethods, realRes) {
-  const routes = routesGenerator(location);
-  const {
-    matchData,
-    pathType,
-  } = matchingPathData(location);
 
-  if (routes[pathType]) {
-    const {
-      apiRoute,
-      actions,
-      errorMessage,
-      serverParams,
-    } = routes[pathType];
-    const route = routePaths[pathType];
-    Actions.updateLoadingStatus(true);
-    const successCb = (response) => {
-      actions.forEach(action => action(response.data));
-      Actions.updateLastLoaded(location);
-      Actions.updateLoadingStatus(false);
-    };
-    const errorCb = (error) => {
-      Actions.updateLastLoaded(location);
-      Actions.updateLoadingStatus(false);
-      console.error(
-        errorMessage,
-        error,
-      );
-    };
-    if (req) {
-      if (serverParams) serverParams(matchData, req);
-      return new Promise((resolve) => {
-        const res = {
-          redirect: url => realRes.redirect(url),
-          json: (data) => {
-            resolve({ data });
-          },
-        };
-        routeMethods[pathType](req, res);
-      })
-        .then(successCb)
-        .catch(errorCb);
-    }
-    return ajaxCall(apiRoute(matchData, route), successCb, errorCb);
-  }
+// This function is called only on the front end, by the DataLoader, when a location changes.
+// Its sole responsibility is to check if any of the configured paths match
+// the current location, and if so, make an api call and pass the resulting data
+// on. Note that it makes use of the fact that now for every frontend route, the
+// corresponding api route can be found simply by adding /api/
+
+function loadDataForRoutes(location, dispatch) {
+  const { pathname } = location;
+
+  const matchingPath = Object.entries(routes).find(([pathKey, pathValue]) => {
+    const { path } = pathValue;
+    return pathname.match(`${baseUrl}/${path}`);
+  });
+
+  if (!matchingPath) return new Promise(resolve => resolve());
+
+  const pathType = matchingPath[0];
+
+  const errorCb = (error) => {
+    console.error(
+      `Error attempting to make ajax request for ${pathType}`,
+      error,
+    );
+  };
+  if (pathType === 'home') successCb(pathType, dispatch);
```

Comment:
1. I am not sure if we want to do this, since it would clear patron data as well, right?
2. If we do reset, I'd like to avoid specific logic in the `dataLoaderUtil`. I would either 
    a.  Make it more generic, like `if (!matchingPath) { ...reset data...}`, or 
    b. Put in in the relevant component's `componentDidMount`

Reply:
This is changed in the other PR. I'll update it in this PR for less confusion. Sorry about that.